### PR TITLE
Issues1160 Hitchhiker's Guide to Testing

### PR DESCRIPTION
**Pull request by samba2**

### CODE REVIEW

File: factcast-site/documentation-docsy/content/en/Usage/HitchhikersGuide/testing.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Building up *projections* works on both API levels, low-level and Factus. 
+However, to simplify development, the high-level Factus API has [explicit support for this concept]({{< ref "/usage/factus/projections/types">}}).
+{{% /alert %}}
+ 
+### Unit Tests
+
+*Projections* are best tested in isolation, ideally at unit test level. 
+At the end they are classes receiving facts and updating some internal state. 
+As we will see in a moment, a unit test is a perfect fit here. 
+However, as soon as the projection's state is externalized (e.g. [see here]({{< ref "/usage/factus/projections/atomicity/" >}})) this test approach can get challenging.
```
Comment (samba2) [inline factcast-site/documentation-docsy/content/en/Usage/HitchhikersGuide/testing.md:22]: or maybe "...might not be sufficient anymore".

File: factcast-site/documentation-docsy/content/en/Usage/HitchhikersGuide/testing.md [hunk 2]
```diff
@@ -216,150 +204,139 @@ This will make the client application connect to the server without using TLS.
...
 ### The User Emails Projection
 
-Now that we know which events to handle, we can process them in the `UserEmailsProjection`:
+Now that we know which events to handle, we can process them in the Factus based `UserEmailsProjection`:
 
 ```java
 public class UserEmailsProjection extends LocalManagedProjection {
 
-  private final Map<UUID, String> userEmails = new HashMap<>();
+    private Map<UUID, String> userEmails = new HashMap<>();
```
Comment (uweschaefer) [inline factcast-site/documentation-docsy/content/en/Usage/HitchhikersGuide/testing.md:319]: why did you remove final here?

**Comment by uweschaefer** [comment]
closes #1160
